### PR TITLE
add Function.name polyfill

### PR DESCRIPTION
**Pull request by nickdhewitt**

### CODE REVIEW

**Comment by nickdhewitt** [comment]
http://jira.krypt.int/browse/KOD-18

**Comment by sahiga** [comment]
Looks good!

**Comment by makosblade** [comment]
I honestly have no idea what this is doing :(

File: dist/kryptnostic.js [hunk 1]
```diff
@@ -24592,6 +24592,60 @@ module.exports = pako;
...
+
+// Inspect the polyfill-ability of this browser
+var needsPolyfill = !("name" in Function.prototype && "name" in (function x() {}));
+var canDefineProp = typeof Object.defineProperty === "function" &&
+  (function() {
+    var result;
+    try {
+      Object.defineProperty(Function.prototype, "_xyz", {
+        get: function() {
+          return "blah";
```
Comment (makosblade) [inline dist/kryptnostic.js]: This looks like it was left in by accident

File: dist/kryptnostic.js [hunk 2]
```diff
@@ -24592,6 +24592,60 @@ module.exports = pako;
...
+        },
+        configurable: true
+      });
+      result = Function.prototype._xyz === "blah";
```
Comment (makosblade) [inline dist/kryptnostic.js]: Also this

**Comment by makosblade** [comment]
I think it's okay but I'm having a hard time understanding what all the logic is actually for

**Comment by nickdhewitt** [comment]
yeah, quick description is: we're using [object].constructor.name to identify objects in validation, and in creating request types in kryptnostic-js. In IE, there is no Function.name support, bc sadly it was not part of the browser spec. (though, it is for ecma script 6, I believe). I forked someone's polyfill, which checks if the current window supports function.name, and adds it if not. I added this as a bower dependency.

File: dist/kryptnostic.js (hunk 1 above)
Comment (nickdhewitt) [inline dist/kryptnostic.js]: so, I'm not sure what the protocol here is, but I assumed that we update the build when we update source code. This snippet was added per description below with the new bower dependency.

**Comment by makosblade** [comment]
Okay so that makes sense to me. I think I'm just confused on why there's so much logic in canDefineProp. Is it normal that you can't define a property on an object? Or is this more IE madness?

**Comment by nickdhewitt** [comment]
https://github.com/kryptnostic/Function.name

**Comment by makosblade** [comment]
Cool, LGTM

File: dist/kryptnostic.js (hunk 1 above)
Comment (rbuckheit) [inline dist/kryptnostic.js]: drew is correct, in the future, don't update the `dist` files when you commit. they should only be updated by the `bump` script.

**Pull request merged**